dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/validation/field/validators/StandardDateValidator.java

```diff
@@ -36,23 +39,38 @@ public FieldValidationResult validateValue(String value, ValidatableField field,
                                                Map<String, ? extends List<? extends ValidatableField>> fieldIndex) {
         value = value.startsWith("-") ? value.substring(1) : value;
 
-        for (DateTimeFormatter parser : PARSERS) {
-            if (isValidDate(value, parser)) {
+        Object yearOnlyRestriction = params.get(RESTRICT_TO_YEAR_ONLY_KEY);
+        if (yearOnlyRestriction == null) {
+            yearOnlyRestriction = false;
+        }
+        boolean restrictToYearOnly = Boolean.parseBoolean(yearOnlyRestriction.toString());
+        if (restrictToYearOnly) {
+            if (isValidDate(value, YEAR_ONLY_PARSER)) {
                 return FieldValidationResult.ok();
             }
+            return FieldValidationResult.invalid(field, BundleUtil.getStringFromBundle("isNotValidYear",
+                    field.getDatasetFieldType().getDisplayName(), YYYY_FORMAT));
+        } else {
+            for (DateTimeFormatter parser : ALL_PARSERS) {
+                if (isValidDate(value, parser)) {
+                    return FieldValidationResult.ok();
+                }
+            }
+            return FieldValidationResult.invalid(field, BundleUtil.getStringFromBundle("isNotValidDate",
+                    field.getDatasetFieldType().getDisplayName(), YYYY_MM_DD_FORMAT, YYYY_MM_FORMAT, YYYY_FORMAT));
         }
-        return FieldValidationResult.invalid(field, BundleUtil.getStringFromBundle("isNotValidDate",
-                field.getDatasetFieldType().getDisplayName(), YYYY_MM_DD_FORMAT, YYYY_MM_FORMAT, YYYY_FORMAT));
     }
 
     // -------------------- LOGIC --------------------
 
     private boolean isValidDate(String value, DateTimeFormatter parser) {
         try {
             TemporalAccessor parsed = parser.parse(value);
-            int year = parsed.get(ChronoField.YEAR_OF_ERA);
-            if (year > 9999) {
-                return false;
+            if (parsed.isSupported(ChronoField.YEAR_OF_ERA)) {
+                int year = parsed.get(ChronoField.YEAR_OF_ERA);
+                if (year > 9999) {
```

Comment:
I know that this was like that before, but it looks strange, so maybe we could extract that magic number to a constant.

Reply:
Lets change as less as possible.